
# [CVE-2021-37655](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-37655)

## Description

- `TensorFlow is an end-to-end open source platform for machine learning. In affected versions an attacker can trigger a read from outside of bounds of heap allocated data by sending invalid arguments to `tf.raw_ops.ResourceScatterUpdate`. The [implementation](https://github.com/tensorflow/tensorflow/blob/f24faa153ad31a4b51578f8181d3aaab77a1ddeb/tensorflow/core/kernels/resource_variable_ops.cc#L919-L923) has an incomplete validation of the relationship between the shapes of `indices` and `updates`: instead of checking that the shape of `indices` is a prefix of the shape of `updates` (so that broadcasting can happen), code only checks that the number of elements in these two tensors are in a divisibility relationship. We have patched the issue in GitHub commit 01cff3f986259d661103412a20745928c727326f. The fix will be included in TensorFlow 2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3, and TensorFlow 2.3.4, as these are also affected and still in supported range.`

## Cvss Data

- **Access Vector**:
  - LOCAL
- **Base Score**:
  - 4.6

## Scores

- **Exploitability Score**:
  - 3.9
- **Impact Score**:
  - 6.4
- **Base Severity**:
  - MEDIUM

## Other Information

- **Publish Date**:
  - 2021-08-12 21:15:08
- **Vulnerability Status**:
  - Analyzed

## References

- **security-advisories@github.com**: https://github.com/tensorflow/tensorflow/commit/01cff3f986259d661103412a20745928c727326f
- **security-advisories@github.com**: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-7fvx-3jfc-2cpc
